operator/pkg/watch/template_change.go

```diff
@@ -50,7 +51,8 @@ func (h *TemplateChangeHandler) Watch(ctx context.Context) handler.MapFunc {
 			Name:      template.GetName(),
 		}
 		for _, kyma := range kymas.Items {
-			if kyma.Spec.Channel != v1alpha1.Channel(channel) {
+			if kyma.Spec.Channel != channel {
+				// what if the channel belongs to the component only?
```

Comment:
This case is uncovered, good catch. I think I will have to adjust the check to also check for the component condition. I think we can do this from L60 on and just move the channel check down.